### DISCUSSION TITLE
Sunsetting Charm Cloud

### DISCUSSION THREAD

**Discussion by bashbunni** [charmbracelet/charm#297] [Announcements]
To continue innovating in this space and supporting our larger projects, we need to keep our small team focused on the most impactful work. On **29 November 2024**, we will be sunsetting Charm Cloud. At that time we will be removing the servers from production and archiving the project on GitHub. Charm Cloud is self-hostable, so you can back up your data before that time and continue using this project on your own server.

The code will continue to be open source and publicly available. If you love this project and would like to maintain a fork, please do.

[Learn more](https://github.com/charmbracelet/charm/blob/main/docs/self-hosting.md) about self-hosting Charm Cloud.

## Impacted tools

To configure both [Skate](https://github.com/charmbracelet/skate) and [Glow](https://github.com/charmbracelet/glow) to work with your own server, you'll need to update the Charm Cloud client environment variables. Glow historically integrated into the Charm Cloud via a stashing feature, but as of [Skate v1.0.0][skatev1] and [Glow v2.0.0][glowv2] the Charm Cloud integration in both applications has been removed (Skate now operates on a local database: to migrate your data see the [Skate v1.0.0 Release Notes][skatev1]). For those using older versions, you can continue to use Charm Cloud features with your own server. Here are the relevant environment variables:

```go
// Config contains the Charm client configuration.
type Config struct {
  Host        string `env:"CHARM_HOST" envDefault:"cloud.charm.sh"`
  SSHPort     int    `env:"CHARM_SSH_PORT" envDefault:"35353"`
  HTTPPort    int    `env:"CHARM_HTTP_PORT" envDefault:"35354"`
  Debug       bool   `env:"CHARM_DEBUG" envDefault:"false"`
  Logfile     string `env:"CHARM_LOGFILE" envDefault:""`
  KeyType     string `env:"CHARM_KEY_TYPE" envDefault:"ed25519"`
  DataDir     string `env:"CHARM_DATA_DIR" envDefault:""`
  IdentityKey string `env:"CHARM_IDENTITY_KEY" envDefault:""`
}
```
Source: [https://github.com/charmbracelet/charm/blob/main/client/client.go](https://github.com/charmbracelet/charm/blob/main/client/client.go#L28-L37)

Thanks for using the Charm Cloud and please chat us up in [Discord](https://charm.sh/chat) if you have any questions.

[glowv2]: https://github.com/charmbracelet/glow/releases/tag/v2.0.0
[skatev1]: https://github.com/charmbracelet/skate/releases/tag/v1.0.0

---

<a href="https://charm.sh/"><img alt="The Charm logo" src="https://stuff.charm.sh/charm-badge.jpg" width="400"></a>

**Comment by ilyagr** [comment]
Sad day. Charm Cloud & Skate were very useful for quick sharing of e.g. SSH public keys when setting up a new computer or VM, or to substitute for clipboard sharing. Now, I guess I'll need to either not have it or to set up an always-on server.

Thanks for making it, supporting it for free for a while, and sharing the idea! Good luck with your new projects!

**Comment by CovertDuck** [comment]
As @bashbunni said on the Discord server some time ago, **us **contributors** could take over the project and maintain it as a community effort.**

Ping me if you are interested in contributing/managing the project. While I really like this piece of software, I would never find the time to maintain it on my lonesome.

So, the more the merrier! :)

**Comment by ilyagr** [comment]
I discovered https://pipe.pico.sh/ and the pico.sh ecosystem, which can replace some of the uses of Charm cloud for me. There's no private file storage, but exchanging information after authentication via ssh works.

There's also https://patchbay.pub/, which I don't understand as well.
